File: CardsAgainstHumanity.UI/State/Games/Models/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using CardsAgainstHumanity.Application.Interfaces;
using CardsAgainstHumanity.Application.Models;

namespace CardsAgainstHumanity.UI.State.Games.Models
{
    public class Game : IGame
    {
        public string Url { get; set; }

        public string Name { get; set; }

        public string Code { get; set; }

        public int CardCount { get; set; } = 7;

        public IList<Player> Players { get; set; } = new List<Player>();

        public Round CurrentRound { get; set; }

        public IList<Round> PreviousRounds { get; set; } = new List<Round>();

        public bool IsOpen { get; set; } = true;

        public bool IsOver { get; set; }

        public Dictionary<int, int> Score => PreviousRounds?.GroupBy(i => i.WonBy).ToDictionary(i => Players.First(p => p.Id == i.Key).Id, i => i.Count());
    }
}

File: CardsAgainstHumanity.Web/src/types/game.ts
// Type definitions for game models
export interface Player {
  id: number;
  name: string;
  cards: string[];
  score: number;
}

export interface Round {
  prompt: string;
  responses: Record<number, string[]>;
  isRevealed: boolean;
  wonBy: number;
}

export interface Game {
  url: string;
  name: string;
  code: string;
  cardCount: number;
  version: number;
  players: Player[];
  currentRound?: Round;
  previousRounds: Round[];
  isOpen: boolean;
  isOver: boolean;
}

export interface PollingResponse {
  version: number;
  hasUpdate: boolean;
}

export interface ApiError {
  message: string;
  status: number;
}
